Pickup the actual username instead of login name for Github

vrenjith opened the pull request.

vrenjith committed:
Update plugin.py

vrenjith committed:
Update plugin.py

vrenjith committed:
Save authors cache after retrieving contributors and during post-build

vrenjith committed:
Merge branch 'patch-1'

vrenjith committed:
Save authors cache during contributor updates and post-build process

vrenjith committed:
Remove logging statement for saving page authors cache

ojacques commented:
Thank you for this. Would it be possible to fallback to login if name is empty / does not exist?
This will preserve earlier behavior with a nice fallback. 
Ideally, a setting which will allow to choose between name and login (defaulting to login to ensure backward compatibility)

Thanks!

vrenjith committed:
Fix author name retrieval in contributors list

vrenjith commented:
> Thank you for this. Would it be possible to fallback to login if name is empty / does not exist? This will preserve earlier behavior with a nice fallback. Ideally, a setting which will allow to choose between name and login (defaulting to login to ensure backward compatibility)
> 
> Thanks!

* That is a good suggestion. I just added that condition to pick up the old behaviour if the `name` does not exist
* Also I added saving of the cache after every update of the internal `dict`. I see that it saves us because in many cases we are hitting API limits and if we have the cache saved everytime, when we rerun the whole build again, it will skip all the items which we already figured out and start from the next of files and eventually it will succeed. Let me know your comments.

vrenjith committed:
Enable logging for saving page authors cache

vrenjith committed:
Log cached authors usage in get_contributors_to_file method

vrenjith committed:
Add logging for plugin configuration status

vrenjith committed:
Bump version to 2.5.2 in setup.py

vrenjith committed:
Enhance logging to include full path of the page authors cache file when loading

vrenjith committed:
Fix return statement in get_contributors_to_file method to properly return cached authors and last commit date

ojacques commented:
Again, thank you. 
I tried this version, but could not find "name" to be populated in the commit informations. Do you use GitHub or GitLab?

vrenjith commented:
I updated the code again.
